feat: used ubiquity-os-logger for logging

ariesgun opened the pull request:
Resolves #4 

<!--
- You must link the issue number e.g. "Resolves #1234"
- Please do not replace the keyword "Resolves" https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword
-->

gentlementlegen commented on src/handlers/query-wallet.ts:
```diff
@@ -30,15 +31,21 @@ export async function registerWallet(context: Context, body: string) {
   if (!address) {
-    return context.logger.info("Skipping to register a wallet address because both address/ens doesn't exist");
+    const message = "Skipping to register a wallet address because both address/ens doesn't exist";
+    await addCommentToIssue(context, `\`\`\`diff\n# ${message}`);
```
If you use the logger you should not have to manually build these string, use the returned value of the logger instead.

gentlementlegen commented on src/types/context.ts:
```diff
@@ -17,12 +18,13 @@ export interface Context<T extends SupportedEventsU = SupportedEventsU, TU exten
-    debug: (message: unknown, ...optionalParams: unknown[]) => void;
-  };
+  logger: Logs;
+  // {
```
Delete comments

gentlementlegen commented on src/handlers/query-wallet.ts:
```diff
@@ -21,24 +22,29 @@ export async function registerWallet(context: Context, body: string) {
   if (!address) {
-    return context.logger.info("Skipping to register a wallet address because both address/ens doesn't exist");
+    const logMessage = logger.info("Skipping to register a wallet address because both address/ens doesn't exist");
+    await addCommentToIssue(context, logMessage.logMessage.diff);
```
```suggestion
    await addCommentToIssue(context, logger.info("Skipping to register a wallet address because both address/ens doesn't exist").logMessage.diff);
```

you can directly inline it to male it simpler.

gentlementlegen commented:
@ariesgun I am okay with the changes, however the `Jest` testing suite is failing, please fix it first.

gentlementlegen commented on package.json:
```diff
@@ -89,5 +91,6 @@
     ]
-  }
+  },
+  "packageManager": "yarn@1.22.22+sha512.a6b2f7906b721bba3d67d4aff083df04dad64c399707841b7acf00f6b133b7ac24255f2652fa22ae3534329dc6180534e98d17432037ff6fd140556e2bb3137e"
```
```suggestion
```

gentlementlegen commented on tests/main.test.ts:
```diff
@@ -31,7 +34,12 @@ describe("Wallet command tests", () => {
-    const spy = jest.spyOn(console, "log");
+    jest
+      .spyOn(RPCHandler.prototype, "getFirstAvailableRpcProvider")
+      .mockResolvedValue(new JsonRpcProvider({ url: "https://fastethrpc.com", skipFetchSetup: true }, Number(1)));
```
I think it would make more sense to just use `passthrough` from `mwjs` and actually use our real RPC package if possible.

ariesgun replied:
Sometimes the test failed because it wasnt able to resolve an ENS. 
https://github.com/ubiquity-os-marketplace/command-wallet/pull/19#issuecomment-2428543283

gentlementlegen replied:
Yes this happens. I guess it's fine for tests, but ideally our RPC shouldn't fail.

gentlementlegen replied:
@0x4007 What do you think would be the best approach? In the context of testing, hardcoding an RPC that we are sure works seems ok, but having our RPCs would be preferable but more unstable.

0x4007 replied:
I'm not sure if its valid to skip implementing this test temporarily until we fix our RPC provider. I plan to work on it on the plane ride all day tomorrow so hopefully I have something to show for it. I might be able to push support for ENS only first and then work on generalizing the encoding/decoding of any contract method.

ariesgun replied:
Shall I undo this change then?

0x4007 replied:
I was notified that some fixes were made to either the RPC manager or one of the interfaces that implements it, but I would always think that an RBC manager would be more stable than a single RPC end point. 

RPC endpoints seem to not have hundred percent up time

ariesgun commented:
I reverted the changes and the tests also pass

> @ariesgun I am okay with the changes, however the `Jest` testing suite is failing, please fix it first.

I reverted the changes and the tests also pass

gentlementlegen commented:
QA: https://github.com/Meniole/command-wallet/issues/5#issuecomment-2436531033
Please provide your own QA as well next time.
